### DISCUSSION TITLE
Multiple trpc endpoints on one client

### DISCUSSION THREAD

**Discussion by NeurAlch** [trpc/trpc#3235] [Q&A] [answered]
Hey! 

I have a new project where I have basically 2 services (subdomains) running independently (both using trpc), I would like the client from one of those to access the other. So server No.1 access trpc from itself and server No. 2. 

What is the best approach to use trpc on the client for multiple endpoints? Ideally I would like something like: `trpc.server1.someQueryOnServer1...`, and `trpc.server2.someQueryOnServer2...`

Found some mixed things searching for Google, but would like to know what is the best approach on V10.

**Comment by NeurAlch** [comment]
Found this: https://github.com/trpc/trpc/pull/2986/commits/3167df4804a4cd36da6142356c6ef96fed6a44a1#diff-b3cf9519d0a70387ebb2bd980aad186f462e834b1ae6c453005864446bc447ec

Currently implementing this, leaving question open for now just in case there is a better known solution I'm missing.

**Comment by Eduard-Hasa** [comment, marked as answer]
https://github.com/trpc/trpc/pull/2986 I got this working with cloudflare workers just fine. I seen someone on github have a version with aws lambdas.

**Reply by NeurAlch** [reply]
Yep thanks, those are the ones I've found so far. Giving it a go with those examples.

**Comment by NeurAlch** [comment]
#2986 Worked for me, I still have some type issues to solve but seems like everything works fine. I also had to update CORS on my servers and the fetch on `httpBatchLink` to use:

```
export const trpcFetchWithCreds: typeof fetch = (url, options) => {
  return fetch(url, {
    ...options,
    credentials: 'include',
  });
};
```

Only thing I have left is to fix typings on the `trcp.Provider` client:

```
TS2740: Type 'DecoratedProcedureRecord<{ ...is missing the following properties from type 'TRPCClient<CreateRouterInner<RootConfig<{ ctx: { req: Request<ParamsDictionary, any, any, ParsedQs, Record<string, any>>; res: Response<any, Record<string, any>>; session: Session & Partial<...>; }; meta: {}; errorShape: DefaultErrorShape; transformer: DefaultDataTransformer; }>, { ...; }>>': links, runtime, requestId, $request, and 4 more
```

**Reply by NeurAlch** [reply]
The type error was my fault, the example uses vanilla, but I have `createTRPCReact`
